.github/workflows/validate.yml

```diff
@@ -2,20 +2,21 @@ name: validate
 on:
   push:
     branches:
-      - '+([0-9])?(.{+([0-9]),x}).x'
-      - 'main'
-      - 'next'
-      - 'next-major'
-      - 'beta'
-      - 'alpha'
-      - '!all-contributors/**'
+      - "+([0-9])?(.{+([0-9]),x}).x"
+      - "main"
+      - "next"
+      - "next-major"
+      - "beta"
+      - "alpha"
+      - "!all-contributors/**"
   pull_request: {}
 jobs:
   main:
     # ignore all-contributors PRs
     if: ${{ !contains(github.head_ref, 'all-contributors') }}
     strategy:
       matrix:
+        eslint: [6.8.0, 6, 7.0.0, 7]
```

Comment:
Is there a reason why we're testing against such specific versions? I would think it would be better if we tested against the min versions needed, but still allow higher versions (e.g. 6.8 rather than 6.8.0)

Reply:
The minimum version needed is `6.8.0`, so that's why I choose to test against that
This way we know for sure that it won't break.

But seems it does break currently 🤔

Reply:
Yeah constraining to a minor version is fine, but that shouldn't mean we need to constraint to a patch - overall what I'm saying is that when you provide a dependency version like this npm will install that _exact_ version (even though it writes to `package.json` with a `^` constraint), so with this we'd only be testing against `6.8.0`, not `6.8.0 or higher`.

What you've got now looks good though, since having the major versions of each will give us the latest of each 👍

Reply:
CI still seems to fail somehow on the older versions though 🤔

Any idea how I can fix that?

Reply:
hmmm I want to say it's maybe due to npm v7+ handling peer dependencies differently, but can't reproduce this locally (e.g. using `--force` does result in v6 of eslint being installed anyway).

I think let's stick an `npm ls --all` after installing eslint to see what our tree looks like - hopefully that'll provide a bit of info. Afterwards we can try seeing if `--legacy-peer-dependencies` makes any difference, since that's an easy one to try as well.

Reply:
I can't see anything wrong 🤔

https://github.com/testing-library/eslint-plugin-jest-dom/runs/4524813794#step:7:1

Reply:
That's weird - we've got some [`eslint` v7 in our tree](https://github.com/testing-library/eslint-plugin-jest-dom/runs/4524813794?check_suite_focus=true#step:7:1316) for some reason :/

Reply:
It seems like `eslint-config-kentcdodds` is pulling in v7, which would be why CI is failing because we're calling a script provided by that package (so it's the closest eslint when it does a `require`)

https://github.com/testing-library/eslint-plugin-jest-dom/runs/4524813794?check_suite_focus=true#step:7:1537

Reply:
I think lets try with `--legacy-peer-dependencies` next (btw probably good to stick a `|| true` on your `npm ls` call to clear it's error code so it doesn't fail on that step)

Reply:
Oh yeah `eslint-config-kentcdodds` has ESLint 7 as it's peerDependency

https://github.com/kentcdodds/eslint-config-kentcdodds/blob/12fb072b9826928ea65eae4d9b1c1ceb32e8b1a3/package.json#L66-L69

Reply:
Seems like `eslint-remote-tester` also needs ESLint v7

https://github.com/AriPerkkio/eslint-remote-tester/blob/1653eba42b43192ed6752a98f2780f505b7ef49b/package.json#L80-L82

Reply:
CI is still failing with `--legacy-peer-dependencies` though 🤔😕

https://github.com/testing-library/eslint-plugin-jest-dom/runs/4525638283?check_suite_focus=true#step:8:17

Reply:
Yeah but it's not the only one with it as a `peerDependency`, so it's double weird it's the only one having this problem.

Frankly I suspect this yet another bug in npm 7 (while I love a lot of what npm 7 has, I've hit a bunch of weird not-really-edge-case bugs like this that really feel like they shouldn't be happening in a stable package manager...).

I wonder maybe if doing an `npm un eslint` before force installing might help somehow 🤔 

(that's a random shot in the dark - I'll try and think of other things we might be able to do)

Reply:
something else we could try just for the sake of it is removing `eslint` using [this script](https://github.com/G-Rath/scripts/blob/master/src/npm_remove_package_from_lock.js), as that should force npm to visit all the parts of the tree that need eslint anyway.

Reply:
all good - I'll try and have a crack at it this weekend.

Reply:
Hi guys I'm eagerly waiting for this update (and all the ones it blocks 😛 ) so I decided to check it out and see if I could help.

I cloned this branch and ran the same commands as the workflow (to install eslint 6.8). Then I ran `npm ls eslint@7` and got this output:
```
└─┬ kcd-scripts@11.2.2
  ├─┬ eslint-config-kentcdodds@19.2.0
  │ ├─┬ @babel/eslint-parser@7.16.5
  │ │ └── eslint@7.32.0 deduped
  │ ├─┬ @babel/eslint-plugin@7.16.5
  │ │ └── eslint@7.32.0 deduped
  │ ├─┬ eslint-config-prettier@8.3.0
  │ │ └── eslint@7.32.0 deduped
  │ ├─┬ eslint-plugin-testing-library@4.12.4
  │ │ └── eslint@7.32.0 deduped
  │ └── eslint@7.32.0 deduped
  └── eslint@7.32.0
```
If you check the repo for kcd-scripts it indeed includes `eslint` in it's [dependencies](https://github.com/kentcdodds/kcd-scripts/blob/fec3fea6dc6d23b62376ed947d75b4fb78b3d01e/package.json#L66). If you remove eslint from `node_modules/kcd-scripts/node_modules` it runs with eslint 6.8 but other errors come up.

IMO opinion it's clear that `kcd-scripts` is not made to be ran with other eslint versions than the one it bundles. So I think a choice needs to be made between using `kcd-scripts` for linting or testing with multiple `eslint` versions

Reply:
🤦 that was my first thought and checked in the package.json but apparently I'm blind.

That is 💯 what is going on here - @MichaelDeBoey are you happy to pick this back up now that we've confirmed the issue?

Reply:
@G-Rath The question stays the same though:
- Do we remove support for ESLint 6?
- Do we remove `kcd-scripts` (which is used for building the package too)?
- Do we stop linting using `kcd-scripts`/`eslint-config-kentcdodds`?

Reply:
@MichaelDeBoey I think we should try and see if we can support v6 without too much extra work for now: this is just for testing against v6, so we don't need `kcd-scripts` to be completely working, just enough to be able to run 

> it runs with eslint 6.8 but other errors come up.

It would be good to know what the actual errors were, since they could be something we could workaround.

(I'm not super attached to v6, but e.g. we've only just started moving projects onto it at work so think it could be good to keep compatibility for even just a few more months until the new year)

Reply:
Just manually ran the workflow to install ESLint 6.8 and deleted `node_modules/kcd-scripts/node_modules/eslint`.
When running `npm run validate`, I get the following output:

<details>
  <summary>console output</summary>

```sh
$ npm run validate

> eslint-plugin-jest-dom@0.0.0-semantically-released validate
> kcd-scripts validate

[build] Successfully compiled 15 files with Babel (1840ms).
[build] npm run build --silent exited with code 0
[test]  PASS  src/__tests__/lib/rules/prefer-focus.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-text-content.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-attribute.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-style.js
[test]  PASS  src/__tests__/lib/rules/prefer-empty.js
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/build/generate-readme-table.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/smoke-test/eslint-remote-tester.config.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/__fixtures__/createBannedAttributeTestCases.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/index.test.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/no-attribute-checking.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   6:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-empty.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-focus.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
  1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   6:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-in-document.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   10:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-prefer-to-have-class.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   1:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-attribute.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-style.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   1:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-text-content.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-value.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/assignment-ast.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/createBannedAttributeRule.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/index.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/queries.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-checked.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-empty.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-enabled-disabled.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
  1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-focus.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-in-document.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-required.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-to-have-attribute.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-to-have-class.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-to-have-style.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-to-have-text-content.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] /Users/michaeldeboey/Developer/eslint-plugin-jest-dom/src/rules/prefer-to-have-value.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
  1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] ✖ 205 problems (205 errors, 0 warnings)
[lint] 
[test]  PASS  src/__tests__/index.test.js
[lint] npm run lint --silent exited with code 1
--> Sending SIGTERM to other processes..
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-value.js
[test]  PASS  src/__tests__/lib/rules/no-attribute-checking.js
[test] npm run test --silent -- --coverage exited with code SIGTERM
```

</ details>

Seems like all these linting rules were introduced in ESLint v7:
- https://eslint.org/docs/rules/no-unsafe-optional-chaining#version
- https://eslint.org/docs/rules/no-loss-of-precision#version
- https://eslint.org/docs/rules/no-unreachable-loop#version
- https://eslint.org/docs/rules/no-useless-backreference#version
- https://eslint.org/docs/rules/default-case-last#version
- https://eslint.org/docs/rules/no-nonoctal-decimal-escape#version
- https://eslint.org/docs/rules/id-denylist#version

Reply:
I think the overall problem is that this flow is meant for taking care of everything, when we're wanting to just run tests - so we should probably keep this job as is, and make a new one that runs just the tests with the matrix of eslint versions.

Reply:
Just manually ran the workflow to install ESLint 6.8 and deleted `node_modules/kcd-scripts/node_modules/eslint`.
When running `npm run validate`, I get the following output:

<details>
  <summary>console output</summary>

```sh
$ npm run validate

> eslint-plugin-jest-dom@0.0.0-semantically-released validate
> kcd-scripts validate

[build] Successfully compiled 15 files with Babel (1840ms).
[build] npm run build --silent exited with code 0
[test]  PASS  src/__tests__/lib/rules/prefer-focus.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-text-content.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-attribute.js
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-style.js
[test]  PASS  src/__tests__/lib/rules/prefer-empty.js
[lint] 
[lint] .../eslint-plugin-jest-dom/build/generate-readme-table.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/smoke-test/eslint-remote-tester.config.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/__fixtures__/createBannedAttributeTestCases.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/index.test.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/no-attribute-checking.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   6:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-empty.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-focus.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
  1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   6:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-in-document.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   10:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-prefer-to-have-class.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   1:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-attribute.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-style.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   1:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-text-content.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
[lint]    1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/__tests__/lib/rules/prefer-to-have-value.js
[lint]    1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found                           no-unsafe-optional-chaining
[lint]    1:1  error  Definition for rule 'no-loss-of-precision' was not found                                  no-loss-of-precision
   1:1  error  Definition for rule 'no-unreachable-loop' was not found                                   no-unreachable-loop
[lint]    1:1  error  Definition for rule 'no-useless-backreference' was not found                              no-useless-backreference
[lint]    1:1  error  Definition for rule 'default-case-last' was not found                                     default-case-last
[lint]    1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found                            no-nonoctal-decimal-escape
[lint]    1:1  error  Definition for rule 'id-denylist' was not found                                           id-denylist
[lint]   11:1  error  'eslint' should be listed in the project's dependencies. Run 'npm i -S eslint' to add it  import/no-extraneous-dependencies
[lint] 
[lint] .../eslint-plugin-jest-dom/src/assignment-ast.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/createBannedAttributeRule.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/index.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/queries.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-checked.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-empty.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-enabled-disabled.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
  1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-focus.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-in-document.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-required.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-to-have-attribute.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-to-have-class.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-to-have-style.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-to-have-text-content.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
[lint]   1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] .../eslint-plugin-jest-dom/src/rules/prefer-to-have-value.js
[lint]   1:1  error  Definition for rule 'no-unsafe-optional-chaining' was not found  no-unsafe-optional-chaining
[lint]   1:1  error  Definition for rule 'no-loss-of-precision' was not found         no-loss-of-precision
[lint]   1:1  error  Definition for rule 'no-unreachable-loop' was not found          no-unreachable-loop
  1:1  error  Definition for rule 'no-useless-backreference' was not found     no-useless-backreference
[lint]   1:1  error  Definition for rule 'default-case-last' was not found            default-case-last
[lint]   1:1  error  Definition for rule 'no-nonoctal-decimal-escape' was not found   no-nonoctal-decimal-escape
[lint]   1:1  error  Definition for rule 'id-denylist' was not found                  id-denylist
[lint] 
[lint] ✖ 205 problems (205 errors, 0 warnings)
[lint] 
[test]  PASS  src/__tests__/index.test.js
[lint] npm run lint --silent exited with code 1
--> Sending SIGTERM to other processes..
[test]  PASS  src/__tests__/lib/rules/prefer-to-have-value.js
[test]  PASS  src/__tests__/lib/rules/no-attribute-checking.js
[test] npm run test --silent -- --coverage exited with code SIGTERM
```

</details>

Seems like all these linting rules were introduced in ESLint v7:
- https://eslint.org/docs/rules/no-unsafe-optional-chaining#version
- https://eslint.org/docs/rules/no-loss-of-precision#version
- https://eslint.org/docs/rules/no-unreachable-loop#version
- https://eslint.org/docs/rules/no-useless-backreference#version
- https://eslint.org/docs/rules/default-case-last#version
- https://eslint.org/docs/rules/no-nonoctal-decimal-escape#version
- https://eslint.org/docs/rules/id-denylist#version

Reply:
@G-Rath I've updated `validate` workflow to only run `lint` against ESLint v7, so I think this should be good now 🙏

Reply:
CI is green now 🥳🎉